src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -156,7 +156,7 @@ internal static BoundStatement BindUsingStatementOrDeclarationFromParts(SyntaxNo
             // In the future it might be better to have a separate shared type that we add the info to, and have the callers create the appropriate bound nodes from it
             if (isUsingDeclaration)
             {
-                return new BoundUsingLocalDeclarations(syntax, disposeMethodOpt, iDisposableConversion, awaitOpt, declarationsOpt, hasErrors);
+                return new BoundUsingLocalDeclarations(syntax, disposeMethodOpt, iDisposableConversion, awaitOpt, originalBinder, declarationsOpt, hasErrors);
```

Comment:
>originalBinder [](start = 114, length = 14)

Could this be an SemanticModel binder (an IncrementalBinder)? We probably don't want to cache it in the tree and should skip to the "regular" binder. Should probably check other places where we cache binders in bound tree.

Reply:
Looks like a couple of other `BoundNode` store binders: `BoundCallExpression` and `BoundObjectInitializerMember` for example. These end up storing the `Binder` value `this` which is an arbitrary sub-class of `Binder`. 

Can you help me understand why that is safe but storing `originalBinder` is not?

Reply:
> Can you help me understand why that is safe but storing originalBinder is not? 

I said: "Should probably check other places where we cache binders in bound tree." And those are good examples to check. Also, I am not sure if "unsafe" is the right term to use here. It is probably unlikely to find an observable negative effect from caching an IncrementalBinder besides holding onto an object that we expect to be transient and short lived. An IncrementalBinder also "modifies" the state of SemanticModel it is associated with. Depending on the usage of the cached binder, that might never happen, or if it happens, this might be desirable or not desirable. By default, I would assume that any effect on SemanticModel is not desirable. There could be exceptions, of course. 

If the sole purpose of capturing the binder is to build arguments later. I think we simply should build the arguments on the spot and capture them instead.

---
In reply to: [539670556](https://github.com/dotnet/roslyn/pull/49842#discussion_r539670556) [](ancestors = 539670556)

Reply:
> If the sole purpose of capturing the binder is to build arguments later. I think we simply should build the arguments on the spot and capture them instead.

Doing this is probably going to affect other nodes/scenarios, better to deal with as a separate change

---
In reply to: [539703344](https://github.com/dotnet/roslyn/pull/49842#discussion_r539703344) [](ancestors = 539703344,539670556)

Reply:
And if we plan to do that later, then probably no reason to worry about skipping binders here now.

---
In reply to: [539704481](https://github.com/dotnet/roslyn/pull/49842#discussion_r539704481) [](ancestors = 539704481,539703344,539670556)

Reply:
See [https://github.com/dotnet/roslyn/issues/49218](https://github.com/dotnet/roslyn/issues/49218)

---
In reply to: [539706285](https://github.com/dotnet/roslyn/pull/49842#discussion_r539706285) [](ancestors = 539706285,539704481,539703344,539670556)